src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -150,7 +153,39 @@ private void OnGetSuggestionsCommandExecute()
                 _suggestionsDropdownTelemetry.DropdownButtonClickTimes += 1;
             }
 
-            _getSuggestionsTask = _smartRenameSession.GetSuggestionsAsync(_cancellationTokenSource.Token).CompletesAsyncOperation(listenerToken);
+            var allRenameLocations = BaseViewModel.Session.AllRenameLocationsTask.Join();
+            ImmutableDictionary<string, string[]> context = default;
+            if (allRenameLocations.Locations.Count > 0)
+            {
+                var references = new List<string>();
+                foreach (var renameLocation in allRenameLocations.Locations)
+                {
+                    var syntaxTree = renameLocation.Document.GetSyntaxTreeSynchronously(_cancellationTokenSource.Token);
+                    var text = syntaxTree.GetText(_cancellationTokenSource.Token);
+                    var lineSpan = syntaxTree.GetLineSpan(renameLocation.TextSpan, _cancellationTokenSource.Token);
+
+                    var startLine = lineSpan.StartLinePosition.Line;
+                    var endLine = lineSpan.EndLinePosition.Line;
+
+                    var documentContent = string.Join(
+                        Environment.NewLine,
+                        text.Lines.Skip(startLine).Take(endLine - startLine + 1).Select(l => l.ToString()));
+                    references.Add(documentContent);
+                }
+                var contextBuilder = ImmutableDictionary.CreateBuilder<string, string[]>();
+                contextBuilder.Add("Reference", references.ToArray());
+                context = contextBuilder.ToImmutableDictionary();
+            }
+            _smartRenameSession.PromptOverride = """
+                Your task is to help a software developer improve the identifier name indicated by [NameThisIdentifier]. The existing identifier name is {identifier}
+
+                Use the following information as context:
+
+                {context}
+
+                Given the provided information, generate five suggestions to rename the selected symbol. The suggested name should match the style of similar identifiers in the provided [CODE]. Put the suggestions in a JSON array called SuggestedNames and return the json object only as a response. Do not include any markdown formatting. Here are an example of the RESPONSE format: { ""SuggestedNames"": [""..."", ""..."", ""..."", ""..."", ""...""] }
+                """;
+            _getSuggestionsTask = _smartRenameSession.GetSuggestionsAsync(context, _cancellationTokenSource.Token).CompletesAsyncOperation(listenerToken);
```

Comment:
looks like we're already kicking off a task here.  So we should push the above expensive work into async work that goes before thsi async call.  It would be trivial to make an async helper method here that does hte above, then calls GetSuggestionsAsync.  and you just do `_getSuggestionsTask = MyNewHelperAsync(arg1, arg2, arg3);` etc.

Reply:
That's a good way to spin up async work.

I'd like to be able to make the `GetRenameContextAsync` call *lazily* within the `SmartRenameViewModel`. 
currently, I'm getting the context in `InlineRenameService` because it has access to `IEditorRenameService`:
`var context = await editorRenameService.GetRenameContextAsync(renameInfo, cancellationToken).ConfigureAwait(false);`
then, I pass it into `InlineRenameSession` as a constructor parameter.

How would you access `IEditorRenameService` from the `RenameFlyoutViewModel`? I don't even have access to `Document` or `renameInfo` here, which are necessary to get the context.